tests/subsys/bluetooth/mesh/scheduler_model/message_validity/src/main.c

```diff
@@ -0,0 +1,554 @@
+	return 0;
+}
+
+int _bt_mesh_time_srv_update_handler(struct bt_mesh_model *model)
+{
+	return 0;
+}
+
+bool bt_mesh_msg_ack_ctx_match(const struct bt_mesh_msg_ack_ctx *ack, uint32_t op, uint16_t addr,
+			       void **user_data)
+{
+	return false;
+}
+
+/* Call server SET opcode handler */
+static int action_set(const struct bt_mesh_schedule_entry *test_entry, bool ack)
+{
+	BT_MESH_MODEL_BUF_DEFINE(
+		buf, ack ? BT_MESH_SCHEDULER_OP_ACTION_SET : BT_MESH_SCHEDULER_OP_ACTION_SET_UNACK,
+		BT_MESH_SCHEDULER_MSG_LEN_ACTION_SET);
+
+	scheduler_action_pack(&buf, test_idx, test_entry);
+
+	return _bt_mesh_scheduler_setup_srv_op[ack ? 0 : 1].func(mock_mod_srv, NULL, &buf);
+}
+
+/* Call server GET opcode handler */
+static int current_register_get(void)
+{
+	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCHEDULER_OP_GET, BT_MESH_SCHEDULER_MSG_LEN_GET);
+
+	return _bt_mesh_scheduler_srv_op[0].func(mock_mod_srv, NULL, &buf);
+}
+
+/* Call server ACTION GET opcode handler */
+static int action_get(void)
+{
+	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCHEDULER_OP_ACTION_GET,
+				 BT_MESH_SCHEDULER_MSG_LEN_ACTION_GET);
+
+	net_buf_simple_add_u8(&buf, test_idx);
+
+	return _bt_mesh_scheduler_srv_op[1].func(mock_mod_srv, NULL, &buf);
+}
+
+/* Send an acked and an unacked SET message with expected return value for both. */
+static void cli_set_msg_send_assert(const struct bt_mesh_schedule_entry *test_entry, int ret_val)
+{
+	zassert_equal(bt_mesh_scheduler_cli_action_set(&sched_cli, NULL, test_idx, test_entry,
+						       NULL),
+		      ret_val);
+	zassert_equal(bt_mesh_scheduler_cli_action_set_unack(&sched_cli, NULL, test_idx,
+							     test_entry),
+		      ret_val);
+}
+
+static void *setup(void)
+{
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.init, "Server init cb is null");
+	zassert_not_null(_bt_mesh_scheduler_cli_cb.init, "Client init cb is null");
+
+	_bt_mesh_scheduler_srv_cb.init(mock_mod_srv);
+	_bt_mesh_scheduler_cli_cb.init(mock_mod_cli);
+
+	return NULL;
+}
+
+static void run_before(void *f)
+{
+	test_idx = 0;
+	msg_sent_cnt = 0;
+}
+
+static void teardown(void *f)
+{
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.reset, "Server reset cb is null");
+	zassert_not_null(_bt_mesh_scheduler_cli_cb.reset, "Client reset cb is null");
+
+	_bt_mesh_scheduler_srv_cb.reset(mock_mod_srv);
+	_bt_mesh_scheduler_cli_cb.reset(mock_mod_cli);
+}
+
+ZTEST_SUITE(scheduler_message_validity_test, NULL, setup, run_before, NULL, teardown);
+
+ZTEST(scheduler_message_validity_test, test_cli_set_invalid)
+{
+	/** (1) Attempt to send a SET and SET_UNACK message with an invalid schedule entry,
+	 * meaning it contains one or more member values that could not be packed without
+	 * compromising its integrity, and check that the client returns with error.
+	 * (2) Verify that no message is sent.
+	 */
+
+	struct bt_mesh_schedule_entry test_entry[5];
+
+	for (size_t i = 0; i < ARRAY_SIZE(test_entry); i++) {
+		test_entry[i] = valid_entry;
+	}
+
+	/* Invalid member values */
```

Comment:
There are 10 fields in scheduler register. I guess customer is able to provide anything in any of them. Why the only 5 are checked?

Reply:
For the others the entire range of the field type used in bt_mesh_schedule_entry is valid according to spec, so it's physically impossible to provide an invalid value, as far as I can tell.

Reply:
For packed form that is traveling around the network - yes. For `struct bt_mesh_schedule_entry` that is incoming data for scheduler client - no. Exactly scheduler client should convert incoming structure into packed data but preliminary it should check correctness of incoming data.